### PR TITLE
Chore: 685 - CI for Mobile apps

### PR DESCRIPTION
**Pull request by bradmartin**
Testing GH action to replace Travis for CI

### CODE REVIEW

**Comment by bradmartin** [comment]
Last 2 things I believe we need to finalize are the SSH Keys that are being used for the private deps. Our `private-keys` repo is configured correctly. I must have uploaded the sqlite repo keys in the incorrect format bc ssh-add fails passing them in from the action.

The authors of the action added multiple SSH key option which we are currently using on this branch https://github.com/webfactory/ssh-agent/pull/14 which will solve that issue once the format is correct on our keys uploaded to the repos.

Last, need to increase the JS Heap to accommodate the large TS project we run when building the apps.

**Comment by bradmartin** [comment]
closes #685

**Comment by finger563** [comment]
building this branch now :+1:

**Comment by finger563** [comment]
Do I *need* to update node or other deps? getting this weird error during the `setup` for `pt.m`:

```
<--- Last few GCs --->

[24606:0x3b90920]     7946 ms: Mark-sweep 1390.8 (1424.4) -> 1390.2 (1423.4) MB, 294.6 / 0.0 ms  (average mu = 0.127, current mu = 0.056) allocation failure scavenge might not succeed
[24606:0x3b90920]     7950 ms: Scavenge 1391.0 (1423.4) -> 1390.4 (1423.9) MB, 2.3 / 0.0 ms  (average mu = 0.127, current mu = 0.056) allocation failure 
[24606:0x3b90920]     7954 ms: Scavenge 1391.2 (1423.9) -> 1390.6 (1424.4) MB, 2.2 / 0.0 ms  (average mu = 0.127, current mu = 0.056) allocation failure 


<--- JS stacktrace --->

==== JS stack trace =========================================

    0: ExitFrame [pc: 0x21d21935be1d]
Security context: 0x17d74e99e6e1 <JSObject>
    1: /* anonymous */(aka /* anonymous */) [0x278520aafbd9] [/home/bob/repositories/permobil-client/node_modules/typescript/lib/tsc.js:~26707] [pc=0x21d21937e338](this=0x2fc4bed826f1 <undefined>,grandchild=0x31e83d1be279 <Node map = 0x39699e605069>)
    2: visitNodes(aka visitNodes) [0x77156aedd1] [/home/bob/repositories/permobil-client/node_modules/typescr...

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0x8dc1c0 node::Abort() [node]
 2: 0x8dc20c  [node]
 3: 0xad60ae v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [node]
 4: 0xad62e4 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [node]
 5: 0xec3972  [node]
 6: 0xec3a78 v8::internal::Heap::CheckIneffectiveMarkCompact(unsigned long, double) [node]
 7: 0xecfb52 v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [node]
 8: 0xed0484 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [node]
 9: 0xed30f1 v8::internal::Heap::AllocateRawWithRetryOrFail(int, v8::internal::AllocationSpace, v8::internal::AllocationAlignment) [node]
10: 0xe9c574 v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationSpace) [node]
11: 0x113beae v8::internal::Runtime_AllocateInNewSpace(int, v8::internal::Object**, v8::internal::Isolate*) [node]
12: 0x21d21935be1d 

```

File: apps/wear/smartdrive/package.json [hunk 1]
```diff
@@ -3,39 +3,39 @@
...
     "nativescript-permissions": "~1.3.8",
     "nativescript-sentry": "file:../../../forked-plugins/nativescript-sentry-1.9.1.tgz",
     "nativescript-sqlite": "~2.3.3",
-    "nativescript-sqlite-commercial": "github:autochair/nativescript-sqlite-commercial",
-    "nativescript-sqlite-encrypted": "github:autochair/nativescript-sqlite-encrypted",
+    "nativescript-sqlite-commercial": "github:autochair/nativescript-sqlite-commercial#master",
+    "nativescript-sqlite-encrypted": "github:autochair/nativescript-sqlite-encrypted#master",
     "nativescript-theme-core": "~1.0.6",
     "nativescript-themes": "~2.0.1",
+    "nativescript-toasty": "~2.0.1",
```
Comment (finger563) [inline apps/wear/smartdrive/package.json]: why was this dependency added?

File: apps/wear/smartdrive/package.json [hunk 2]
```diff
@@ -3,39 +3,39 @@
...
     "date-fns": "~1.30.1",
     "injection-js": "~2.2.2",
     "lodash": "~4.17.15",
     "nativescript-android-sensors": "~1.5.0",
-    "nativescript-animated-circle": "file:../../../forked-plugins/nativescript-animated-circle",
+    "nativescript-animated-circle": "file:../../../forked-plugins/nativescript-animated-circle-1.3.0.tgz",
     "nativescript-appavailability": "~1.3.2",
     "nativescript-auto-fit-text": "file:../../../forked-plugins/nativescript-auto-fit-text-1.0.1.tgz",
     "nativescript-bluetooth": "file:../../../forked-plugins/nativescript-bluetooth",
-    "nativescript-download-progress": "~1.2.0",
```
Comment (finger563) [inline apps/wear/smartdrive/package.json]: why was this dependency removed?

File: .github/workflows/builds.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          sudo pip install six
+
+      - name: Setup Permobil-Client
+        run: |
+          echo no | npm i -g nativescript
+          tns usage-reporting disable
+          tns error-reporting disable
+          npm run global.installs
+          npm i
+          npm run setup.forked.plugins && npm run setup.mobile && npm run setup.@permobil
```
Comment (finger563) [inline .github/workflows/builds.yml]: probably want to make sure we run `setup.wear` as well

File: apps/wear/smartdrive/package.json (hunk 1 above)
Comment (bradmartin) [reply apps/wear/smartdrive/package.json]: Copy error cleaning the PR

File: apps/wear/smartdrive/package.json (hunk 2 above)
Comment (bradmartin) [reply apps/wear/smartdrive/package.json]: Same as above. Copy error cleaning the PR

File: .github/workflows/builds.yml (hunk 3 above)
Comment (bradmartin) [reply .github/workflows/builds.yml]: Yeah that'll be next PR to get wear apps working. Which will be simple running nuki and we drop all those cmds. Nuki isn't used until I get the sqlite private repos SSH solved for the CI. Working on that once we get SD.W building again to merge this

**Pull request merged**